### DISCUSSION TITLE
Is it possible to build a service on AWS using self-hosted Supabase?

### DISCUSSION THREAD

**Discussion by itskey-rsa** [supabase/supabase#13202] [Questions]
Is it possible to use self-hosted Supabase to build a service on AWS? 

Isn't self-hosted only intended for testing and development purposes, rather than for use in production? 
I was able to run `supabase start` on an EC2 instance on AWS, so I believe it should be possible to use it without any issues by using WAF and load balancers to secure it.

**Comment by mul14** [comment]
@HayashiItsuki yes, that's possible. I already use Supabase self-hosted on Amazon EKS and Linode LKE. The setup is quite challenging. And not all features are available.

For EC2, I think it would be better if you deploy as Docker container, rather than supabase command line.

Here are references:
- https://github.com/supabase-community/supabase-kubernetes/blob/main/charts/supabase
- https://github.com/supabase/supabase/tree/master/docker

**Reply by tolikkostin** [reply]
Hey man, I would apreciate if you will share how to run Supabase on EKS with their helm chart. Thanks an advance!

**Reply by igorgrin** [reply]
Same here.  I launched it on ec2, but it's not trivial to do it on EKS.  Can you share please?

**Comment by Nikola-Milovic** [comment]
Its definitely possible to self-host, supabase has open-sourced all of their parts (most of them), so you can grab whatever you want and self host it. Supabase hosted version just handles the complexity for you

**Comment by igorgrin** [comment]
@mul14  - How did you do it on EKS?  Can you share?  I finally got all the pods up, but it's not functioning correctly.

> @HayashiItsuki yes, that's possible. I already use Supabase self-hosted on Amazon EKS and Linode LKE. The setup is quite challenging. And not all features are available.
> 
> For EC2, I think it would be better if you deploy as Docker container, rather than supabase command line.
> 
> Here are references:
> 
> * https://github.com/supabase-community/supabase-kubernetes/blob/main/charts/supabase
> * https://github.com/supabase/supabase/tree/master/docker
